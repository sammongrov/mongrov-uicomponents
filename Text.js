import React, { Component } from 'react';
import { StyleSheet, Text as RNText } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  defaultStyle: {
    fontFamily: 'OpenSans-Regular',
  },
});

export default class Text extends Component {
  static propTypes = {
    children: PropTypes.node,
    style: PropTypes.oneOfType([PropTypes.number, PropTypes.shape(), PropTypes.array]),
  };

  static defaultProps = {
    style: null,
    children: null,
  };

  render() {
    const { style, children } = this.props;
    return (
      <RNText {...this.props} style={[styles.defaultStyle, style]}>
        {children}
      </RNText>
    );
  }
}
