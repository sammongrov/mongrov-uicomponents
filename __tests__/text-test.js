import React from 'react';
import renderer from 'react-test-renderer';
// import { Alert } from 'react-native';
import Text from '../Text';

// const onHelpPress = () => Alert.alert('Help');

it('Text renders correctly', () => {
  const tree = renderer.create(<Text> hello </Text>).toJSON();
  expect(tree).toMatchSnapshot();
});
