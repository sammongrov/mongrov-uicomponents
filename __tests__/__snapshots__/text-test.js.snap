// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "OpenSans-Regular",
      },
      null,
    ]
  }
>
   hello 
</Text>
`;
